CMakeLists.txt

```diff
@@ -35,6 +35,8 @@ ENDIF(POLICY CMP0048)
 include(CheckCXXCompilerFlag)
 CHECK_CXX_COMPILER_FLAG("-std=c++11 " COMPILER_SUPPORTS_CXX11)
 CHECK_CXX_COMPILER_FLAG("-std=c++0x " COMPILER_SUPPORTS_CXX0X)
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ")
+SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS}")
```

Comment:
What are these for?

Reply:
Looking at the branch I had some optimizations and in the conflict removed them. I'm going to think a little more about adding them back in and then make a resolution of keeping these lines with the optimizations or remove the line entirely, thanks!

Reply:
Got it. If we add them back, and I understand correctly how this all is working, then I think we should use distinct vars rather than exact copies of the cmake parameters. Something like CXX_OPTIMIZATION_FLAGS. I do think optimizations are going to be desirable. I almost feel like we need some kind of automated/reproducible way to measure things before getting too fancy with optimizations. I.e. a microbenchmark suite, something to measure bin size, and something to measure memory usage.